core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/CachedBlockQueue.java

```diff
@@ -73,7 +73,7 @@ public void add(CachedBlock cb) {
       heapSize += cb.heapSize();
     } else {
       CachedBlock head = queue.peek();
-      if (cb.compareTo(head) > 0) {
+      if (head != null && cb.compareTo(head) > 0) {
```

Comment:
I'm not sure about this change. If the queue is empty (peek returns null), then this cb is never queued. I'm not sure if this situation is ever possible, unless `maxSize < 0`, though.

Ultimately, I don't see the motivate for changing this behavior.

Reply:
I targeted 2.1.1 because this address issues with the test where they were not testing what was expected - the block size calculations were being ignored. While the function correctly, they were not tested by these tests.  If a future bug fix was identified in 2.1 and implemented, then the updated tests might help so the changes were functioning as expected.

The additional of the null check was to resolve check style identifying that head could be null and cause an NPE. And while it might not occur with the current code, it could in the future.  The other options would be to modify the compareTo to properly handle nulls or use a NonNull annotation (probably preferred) but those seemed more intrusive.

Reply:
I suspect the code was written with the assumption that when in the else that there is always something in the queue and peek will not return null.  So maybe instead of ignoring null, could throw an assertionerror or illegal state exception when seeing null.

Reply:
Modified in d0b5f0d3cb to use Objects.requireNonUnull

Reply:
> Modified in https://github.com/apache/accumulo/commit/d0b5f0d3cba000bc4a05871c413c9023d09892e9 to use Objects.requireNonUnull

I wonder if that will resolve the checkstyle issue.  Can you remove new check in the if statement?

Reply:
bfc8f4f919 - Removed `head != null` check that is not necessary when Objects.requireNonNull is used.